File: src/main/java/org/micoli/php/ui/components/tasks/tree/LabeledTreeNode.java
package org.micoli.php.ui.components.tasks.tree;

import javax.swing.tree.DefaultMutableTreeNode;

public abstract class LabeledTreeNode extends DefaultMutableTreeNode {

    private String label;

    public LabeledTreeNode(Object userObject, String label) {
        super(userObject, true);
        this.label = label;
    }

    protected void setLabel(String label) {
        this.label = label;
    }

    public String getLabel() {
        return this.label;
    }

    public String getTooltip() {
        return this.label;
    }
}

File: src/main/kotlin/org/micoli/php/ui/components/tasks/tree/LabeledTreeNode.kt
package org.micoli.php.ui.components.tasks.tree

import javax.swing.tree.DefaultMutableTreeNode

abstract class LabeledTreeNode : DefaultMutableTreeNode {

    private var innerLabel: String

    constructor(userObject: Any?, label: String) : super(userObject) {
        this.innerLabel = label
    }

    protected fun setLabel(label: String) {
        this.innerLabel = label
    }

    fun getLabel(): String {
        return this.innerLabel
    }

    fun getTooltip(): String {
        return this.innerLabel
    }
}
